sdgym/benchmark.py

```diff
@@ -1080,3 +1218,274 @@ def benchmark_single_table(
+    retrieve them later in the EC2 instance. This is necessary because the
+    EC2 instance does not have access to the local file system of the machine
+    that initiated the run.
+    The pkl file generated will be deleted after the run is completed.
+    """
+    parsed_url = urlparse(output_destination)
+    bucket_name = parsed_url.netloc
+    path = parsed_url.path.lstrip('/') if parsed_url.path else ''
+    job_args_key = f'job_args_list_{str(uuid.uuid4())}.pkl'
+    if path:
+        job_args_key = f'{path}/{job_args_key}' if path else job_args_key
+
+    serialized_data = pickle.dumps(job_args_list)
+    encoded_data = base64.b64encode(serialized_data).decode('utf-8')
+    s3_client.put_object(Bucket=bucket_name, Key=job_args_key, Body=encoded_data)
+
+    return bucket_name, job_args_key
+
+
+def _run_on_aws(output_destination, synthesizers, s3_client, job_args_list):
+    bucket_name, job_args_key = _store_job_args_in_s3(output_destination, job_args_list, s3_client)
+    credentials = s3_client._request_signer._credentials
+    access_key = credentials.access_key
+    secret_key = credentials.secret_key
+    region_name = s3_client.meta.region_name
+
+    # Create script content using just the bucket name
+    script_content = f"""
+import boto3
+import pickle
+import base64
+import pandas as pd
+import sdgym
+from sdgym.synthesizers.sdv import (
+    CopulaGANSynthesizer, CTGANSynthesizer,
+    GaussianCopulaSynthesizer, HMASynthesizer, PARSynthesizer,
+    SDVRelationalSynthesizer, SDVTabularSynthesizer, TVAESynthesizer
+)
+from sdgym.synthesizers import RealTabFormerSynthesizer
+from sdgym.benchmark import _run_jobs, _write_run_id_file, _update_run_id_file
+from io import StringIO
+
+s3_client = boto3.client(
+    's3',
+    aws_access_key_id='{access_key}',
+    aws_secret_access_key='{secret_key}',
+    region_name='{region_name}'
+)
+response = s3_client.get_object(Bucket='{bucket_name}', Key='{job_args_key}')
+encoded_data = response['Body'].read().decode('utf-8')
+serialized_data = base64.b64decode(encoded_data.encode('utf-8'))
+job_args_list = pickle.loads(serialized_data)
+run_id = _write_run_id_file('{output_destination}', {synthesizers}, job_args_list, s3_client)
+scores = _run_jobs(None, job_args_list, False, s3_client)
+_update_run_id_file('{output_destination}', run_id, s3_client)
+s3_client.delete_object(Bucket='{bucket_name}', Key='{job_args_key}')
+"""
+
+    # Create a session and EC2 client using the provided S3 client's credentials
+    session = boto3.session.Session(
+        aws_access_key_id=s3_client._request_signer._credentials.access_key,
+        aws_secret_access_key=s3_client._request_signer._credentials.secret_key,
+        region_name=s3_client.meta.region_name,
+    )
+    ec2_client = session.client('ec2')
+    print(f'This instance is being created in region: {session.region_name}')  # noqa
+
+    # The user data script
+    user_data_script = textwrap.dedent(f"""\
```

Comment:
I think we should split this part out into its own function as we may override it